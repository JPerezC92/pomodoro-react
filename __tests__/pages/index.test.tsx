import { fireEvent, prettyDOM, render, screen } from "@testing-library/react";

import Home from "@/pages/index";

describe("Home ", () => {
  test("renders homepage unchanged", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });

  test("should render a form", () => {
    render(<Home />);

    const form = screen.getByRole("form");

    expect(form).toBeInTheDocument();
  });

  test("should add a new task", () => {
    render(<Home />);

    const form = screen.getByRole("form");
    const taskList = screen.getByRole("list");
    const input = screen.getByPlaceholderText("Add a new task");

    expect(taskList.childElementCount).toBe(0);
    expect(form).toBeInTheDocument();
    expect(input).toBeInTheDocument();

    fireEvent.change(input, { target: { value: "New task" } });

    const button = screen.getByRole("button", { name: "Add task" });

    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent("Add");

    fireEvent.click(button);

    expect(input).toHaveValue("");
    expect(taskList.childElementCount).toBe(1);
  });
});
