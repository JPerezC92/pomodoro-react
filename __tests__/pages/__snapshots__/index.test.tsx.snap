// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home  renders homepage unchanged 1`] = `
<div>
  <div
    class="container"
  >
    Pomodoro
    <form
      name="taskFormCreate"
      role="form"
    >
      <input
        id="title"
        name="title"
        placeholder="Add a new task"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add task
      </button>
    </form>
    <ul />
  </div>
</div>
`;
